ext/standard/php_fopen_wrapper.c

```diff
@@ -323,12 +334,29 @@ php_stream * php_stream_url_wrap_php(php_stream_wrapper *wrapper, const char *pa
 			return NULL;
 		}
 
-		fd = dup((int)fildes_ori);
-		if (fd == -1) {
-			php_stream_wrapper_log_error(wrapper, options,
-				"Error duping file descriptor " ZEND_LONG_FMT "; possibly it doesn't exist: "
-				"[%d]: %s", fildes_ori, errno, strerror(errno));
-			return NULL;
+		if (is_raw) {
+			fd = fildes_ori;
+
+#if defined(PHP_WIN32)
+			if ((HANDLE)_get_osfhandle(fd) == INVALID_HANDLE_VALUE) {
+#elif HAVE_FCNTL_H
+			if (fcntl(fd, F_GETFD) == -1) {
+#else
+#warning "No API to check file descriptor for this operating system, php://fdraw/ will not work"
+			if (0) {
```

Comment:
I'd rather have if(1) here... I.e. if no way to check, then every descriptor is invalid. Otherwise it may be too dangerous?